x/evm/handler.go

```diff
@@ -62,8 +62,10 @@ func handleMsgEthereumTx(ctx sdk.Context, k Keeper, msg types.MsgEthereumTx) (*s
 
 	// Prepare db for logs
 	// TODO: block hash
-	k.CommitStateDB.Prepare(ethHash, common.Hash{}, k.TxCount)
-	k.TxCount++
+	if !st.Simulate {
```

Comment:
why is this the case?

Reply:
the txCount is used in the stateDB, and since a simulated tx is run only on the node it's submitted to, then this will cause the txCount/stateDB of the node that ran the simulated tx to be different than the other nodes, causing the consensus error

Reply:
ahh great catch! I can we document this on the comments?

Reply:
sure I'll open a follow up with comments!